### PR TITLE
Algorithm for Maximum rope cutting problem

### PR DESCRIPTION
**Pull request by atharva20-coder**
### **Describe your change:**

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

#### References
#3591 

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by atharva20-coder** [comment]
Please help me out, I don't know whats wrong here in my system this program runs perfectly

File: src/main/java/com/thealgorithms/dynamicprogramming/MaximumRopeCutting.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+		          Math.max(maxCuts(n-b, a, b, c), 
+		          maxCuts(n-c, a, b, c)));
+		if(res == -1)
+			return -1;
+		return res + 1; 
+	}
+    public static void main(String [] args) 
+    {
+    	int n = 5, a = 2, b = 1, c = 5;
+    	System.out.println(maxCuts(n, a, b, c));
```
Comment (Teuton-Roy) [inline src/main/java/com/thealgorithms/dynamicprogramming/MaximumRopeCutting.java:17]: Can You make it user defined..

Comment (atharva20-coder) [reply src/main/java/com/thealgorithms/dynamicprogramming/MaximumRopeCutting.java:17]: Okay, I will do it

Comment (atharva20-coder) [reply src/main/java/com/thealgorithms/dynamicprogramming/MaximumRopeCutting.java:25]: @Teuton-Roy while adding scanner class it gives error while building

**Review by siriak — CHANGES_REQUESTED** [review]
Please add problem statement, JUnit tests, and remove `main`

**Pull request closed without merging**